### DISCUSSION TITLE
Show feed-in tariffs

### DISCUSSION THREAD

**Discussion by Vaszago** [mampfes/ha_epex_spot#41] [Ideas]
I would like to have the feed-in tariff for excess solar power displayed. Like the feed-in tariff from Awattar Sunny
<img width="538" alt="image" src="https://github.com/mampfes/ha_epex_spot/assets/66181893/0b5252e2-dab6-41ba-81d7-08ba7e6ef0b4">

or E-Control. 
<img width="640" alt="image" src="https://github.com/mampfes/ha_epex_spot/assets/66181893/ba8cfc72-e68c-47e7-a650-f605d9dbb966">

So that I can calculate the electricity costs better.

**Comment by mampfes** [comment]
Sorry - I don't fully understand your request.
Do you want a sensor that provides the "Einspeisetarif" value e.g. 8,405 ct/kWh?

**Comment by Vaszago** [comment]
Yes, I would like a sensor from Awattar or e-control. I think e-control "Marktpreis" is easier to import from phelix-at futures than Awattar. I have already asked Awattar to include the feed-in tariff in the API. But apparently that hasn't happened yet. I don't know if you can extract the import tariff from here automatically. https://docs.google.com/spreadsheets/d/1emzAMIOEhKsddDUaeJkaEbIZTZ2fGLYKAfxGJhuvZrw/edit#gid=474031259

**Comment by riesi77** [comment]
If you open https://www.e-control.at/marktteilnehmer/oeko-energie/marktpreis, you do have an interactive graphics showing the current and past "Einspeisetarife". Benefit would be to even have historic values. But I guess it's not trivial to crawl that and every time they change the structure of the webpage, the integration breaks.
I hope Oemag offers an API for that, soon.

In the meantime, you could just add a number-helper and enter the quaterly value:
![image](https://github.com/mampfes/ha_epex_spot/assets/67387975/cc07be3f-7246-4636-b6b7-1ccd85e0bf32)

**Comment by gerione** [comment]
If you want the eControl price: 
- Make a scrape sensor: 
![image](https://github.com/mampfes/ha_epex_spot/assets/59999828/9c85a4e1-a5be-4944-865d-a75427941a00)
- Add the right ressource: https://www.e-control.at/de/marktteilnehmer/oeko-energie/marktpreis
![image](https://github.com/mampfes/ha_epex_spot/assets/59999828/4f185111-b521-4b8a-8cbd-6294f3c12cbf)
- Sensor: .journal-content-article > table:nth-child(9) > tbody:nth-child(1) > tr:nth-child(8) > td:nth-child(2)
- {{ value | replace (",", ".") | float }}
![image](https://github.com/mampfes/ha_epex_spot/assets/59999828/102378f8-066e-484d-967a-b69e246eb5ad)

**Reply by riesi77** [reply]
That's incredible. Thank you very much for that! Do you know when the sensor is updated / how often the page is polled?

**Reply by gerione** [reply]
I think around once an hour or so. The price was updated today on the homepage and I already have the new price in my sensors. So the last day of the quarter is actually off. But I can live with that.

**Comment by Vaszago** [comment]
Thanks, it worked great.
  
Would this method also be possible for the Awattar site? The price is also shown on the page here.

**Reply by gerione** [reply]
Ressource: https://www.awattar.at/tariffs/sunny
Selector: #wrapper > div.container > div.col-md-9 > div.row.hidden-xs > div > table > tbody > tr:nth-child(1) > td:nth-child(3)
Value template: {{ value.replace(',', '.')|regex_replace('[^.\d]', '') }}

My approach is similar to the following: https://support.didomi.io/how-to-find-the-css-selector-for-your-compliance-report
Copy selector.

**Comment by Vaszago** [comment]
The request worked. How can I display the value in EURO/kWh? It currently shows CENT/kWh. Thanks

**Reply by gerione** [reply]
Value template: {{ value.replace(',', '.')|regex_replace('[^.\d]', '')| float * 10 }}

**Comment by Vaszago** [comment]
Thanks, it worked great

**Comment by aichingerk** [comment]
Hi, is this also possible for the Smart Energy (https://www.smartenergy.at/smartcontrol) API, to become a sensor:
https://apis.smartenergy.at/market/v1/price

**Reply by gerione** [reply]
I guess it could work with this: https://www.home-assistant.io/integrations/sensor.rest/
